test/e2e/test_e2e_dag.py

```diff
@@ -236,9 +235,11 @@ def test_dag_pipeline_creation_and_rerun(self) -> None:
         logging.info(f"First run produced {first_run_entities} unique entities")
 
         # Lookup works too
-        test_key = final_df.filter(
-            pl.col(source_b.f(source_b.config.key_field.name)).is_not_null()
-        )[source_b.f(source_b.config.key_field.name)][0]
+        test_key = list(
+            self.linked_testkit.find_entities(
+                min_appearances={source_b.name: 1, source_a.name: 1}
+            )[0].keys[source_b.name]
+        )[0]
```

Comment:
Whenever I see `list()[0]` I wonder if `next()` could do the job.

```suggestion
        test_key = next(
            self.linked_testkit.find_entities(
                min_appearances={source_b.name: 1, source_a.name: 1}
            )[0].keys[source_b.name]
        )
```

Reply:
Done